src/posthog-core.ts

```diff
@@ -870,6 +880,21 @@ export class PostHog {
             this.setPersonPropertiesForFlags(finalSet)
         }
 
+        if (!isKnownUnEditableEvent(data.event)) {
```

Comment:
ah, you all missed my deliberate mistake 😅 

this stops us processing these events at all... which is correct for sampling but not for masking

so... 

* all events should be processed
* isKnownUnEditableEvent reject changes which are nullish (including empty object)
* others allow those changes but might warn
* empty object should be treated as nullish
* we should still be validating the shape of the events after processing so we can warn if people do something unexpected

Reply:
actually other than $set that still makes sense to me

Reply:
so...

if we say it's not editable then we don't let you edit this at all
you can still use sanitize props

we warn if you remove all properties
$set and $set_once can be undefined so very hard to validate them